src/parquet/column/statistics.h

```diff
@@ -0,0 +1,133 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_COLUMN_STATISTICS_H
+#define PARQUET_COLUMN_STATISTICS_H
+
+#include <cstdint>
+#include <string>
+
+#include "parquet/types.h"
+#include "parquet/schema/descriptor.h"
+#include "parquet/util/mem-allocator.h"
+
+namespace parquet {
+
+class RowGroupStatistics {
+ public:
+  int64_t NullCount() const { return null_count_; }
+  int64_t DistinctCount() const { return distinct_count_; }
+  int64_t NumValues() const { return num_values_; }
+
+  virtual bool HasMinMax() const = 0;
```

Comment:
That's a kind of crappy substitute for C++17 `std::optional` which I avoid for the time being.